homeassistant/components/wake_word/__init__.py

```diff
@@ -88,21 +88,23 @@ def supported_wake_words(self) -> list[WakeWord]:
 
     @abstractmethod
     async def _async_process_audio_stream(
-        self, stream: AsyncIterable[tuple[bytes, int]]
+        self, stream: AsyncIterable[tuple[bytes, int]], wake_word_id: str
     ) -> DetectionResult | None:
         """Try to detect wake word(s) in an audio stream with timestamps.
 
         Audio must be 16Khz sample rate with 16-bit mono PCM samples.
         """
 
     async def async_process_audio_stream(
-        self, stream: AsyncIterable[tuple[bytes, int]]
+        self, stream: AsyncIterable[tuple[bytes, int]], wake_word_id: str | None = None
     ) -> DetectionResult | None:
         """Try to detect wake word(s) in an audio stream with timestamps.
 
         Audio must be 16Khz sample rate with 16-bit mono PCM samples.
         """
-        result = await self._async_process_audio_stream(stream)
+        if wake_word_id is None:
+            wake_word_id = self.supported_wake_words[0].ww_id
```

Comment:
If we do like this we need to:
- Make it a documented requirement that `supported_wake_words` is not allowed to return an empty list
- Document that `supported_wake_words[0]` is the default wake word

Both docstring and [developer documentation](https://developers.home-assistant.io/docs/core/entity/wake_word) should be updated. 

Another option would be to add a new property `default_wake_word_id` or something like that.

Reply:
Addressed